spec.emu

```diff
@@ -1114,6 +1141,969 @@
+          </table>
+        </emu-table>
+      </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-scopes-grammar">
+      <h1>Scopes grammar</h1>
+      <p>The scopes field is described by the following <em>context-sensitive</em> grammar starting at the |Scopes| goal symbol:</p>
+      <emu-grammar type="definition">
+        Scopes :
+          OriginalScopeTreeList
+          TopLevelItemList
+          OriginalScopeTreeList `,` TopLevelItemList
+
+        OriginalScopeTreeList :
+          OriginalScopeTreeItem
+          OriginalScopeTreeList `,` OriginalScopeTreeItem
+
+        OriginalScopeTreeItem :
+          OriginalScopeTree
+          [empty]
+
+        TopLevelItemList :
+          TopLevelItem
+          TopLevelItemList `,` TopLevelItem
+
+        TopLevelItem :
+          GeneratedRangeTree
+          UnknownItem
+      </emu-grammar>
+
+      <p>The number of |OriginalScopeTreeItem|s in the |OriginalScopeTreeList| is expected to match the length of the sources field. The n-th |OriginalScopeTreeItem| describes the scope tree of the n-th authored source file in the sources field. If no scope information is available for a particular source URL, an empty item can be used.</p>
+
+      <emu-note example>
+        <p>A source map with three authored sources where only the first and third source file have scope information available would be encoded as:</p>
+        <pre>
+          "scopes": "&lt;OriginalScopeTree&gt;,,&lt;OriginalScopeTree&gt;,&lt;GeneratedRangeTree&gt;"
+        </pre>
+      </emu-note>
+
+      <emu-clause id="sec-original-scope-grammar">
+        <h1>Original scope grammar</h1>
+        <p>Each original scope consists of:</p>
+        <ul>
+          <li>a |OriginalScopeStart| item describing the start location and general scope information.</li>
+          <li>an optional |OriginalScopeVariablesItem| describing the original names of the variables this scope declares.</li>
+          <li>an optional list of |OriginalScopeTree|s describing child scopes.</li>
+          <li>a |OriginalScopeEnd| item describing the end location.</li>
+        </ul>
+        <p>In grammar form:</p>
+        <emu-grammar type="definition">
+          OriginalScopeTree :
+            OriginalScopeStart OriginalScopeVariablesItem? OriginalScopeItemList? `,` OriginalScopeEnd
+
+          OriginalScopeVariablesItem :
+            `,` OriginalScopeVariables
+
+          OriginalScopeItemList :
+            `,` OriginalScopeItem
+            OriginalScopeItemList `,` OriginalScopeItem
+
+          OriginalScopeItem :
+            OriginalScopeTree
+            UnknownItem
+
+          OriginalScopeStart :
+            `B` ScopeFlags ScopeLine ScopeColumn ScopeNameOrKind? ScopeKind?
+
+          OriginalScopeEnd :
+            `C` ScopeLine ScopeColumn
+
+          OriginalScopeVariables :
+            `D` ScopeVariableList
+
+          ScopeVariableList :
+            ScopeVariable
+            ScopeVariableList ScopeVariable
```

Comment:
I haven't checked yet the SDOs, but it _might_ make sense to have a generic `VlqList` production that we can reuse in multiple places.